service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionService.java

```diff
@@ -19,7 +26,15 @@
 public class CaseSubmissionService {
     private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
     private final UploadDocumentService uploadDocumentService;
-    private final CaseSubmissionGenerationService documentGenerationService;
+    private final IdamClient idamClient;
+    private final RequestData requestData;
+
+    private CaseSubmissionGenerationService documentGenerationService;
+
+    @Autowired
+    public void setCaseSubmissionGenerationService(CaseSubmissionGenerationService documentGenerationService) {
```

Comment:
This seems to solve an error message about a circular dependancy between caseSubmissionService and caseSubmissionGenerationService.

Error I was getting before this change below:

![image](https://user-images.githubusercontent.com/20306332/118476613-a2dc5400-b705-11eb-995e-237564fe71ea.png)

Reply:
was not aware that it requires circular dependencies. You can move getSigneeName to CaseSubmissionGenerationService to avoid it. 

If you want to have it you can use
    @Autowired
    private CaseSubmissionGenerationService documentGenerationService;
    
 setter should not be needed